cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/BaseController.java

```diff
@@ -93,10 +127,39 @@ else if (ex instanceof UnsupportedOperationException) {
             // no need for returning different exit codes for different exceptions, but that's also possible in the
             // future
             return 1;
+        });
+
+        // parse args now to be able to configure logging before we continue running the rest of the CLI
+        // a nice catch of this approach: we can properly handle all sorts of argument parsing errors nicely
+        try {
+            cli.parseArgs(args);
+        } catch (CommandLine.PicocliException e) {
```

Comment:
I think you can use Exception e instead of CommandLine.PicocliException e, 
because the PicocliException extends RuntimeException and the name of the exception is a little confusing.

Reply:
Not sure if we should just catch all exceptions and log them there. I intended to only add some special handling for picocli stuff.
The `PicocliException` is the "base" of all other exceptions in picocli.
If you think we should add a catch-all, that's fine with me, too.